### PR TITLE
Added --authorizedkeys handler

### PR DESCRIPTION
**Pull request by Duologic**
Checks which authorized keys are on a host.

### CODE REVIEW

**Comment by jonathanslenders** [comment]
Sorry, not going to merge this.
This is functionality that can easily be layered on top of the deployer library. It should not be part of this library.

The `RunNode` you have here can be placed in your usual deployment tree. You add all the hosts there. I would also inherit from `ParallelNode` instead of `Node`.
There are a few suggestions I made in the comments.

File: deployer/shell.py [hunk 1]
```diff
@@ -357,6 +357,65 @@ class RunWithSudo(Run):
...
+                keytypes = {'ecdsa-sha2-nistp256',
+                            'ecdsa-sha2-nistp384',
+                            'ecdsa-sha2-nistp521',
+                            'ssh-dss',
+                            'ssh-rsa'
+                           }
+
+                for host in self.hosts:
+                    shell.write('\n\nAuthorized keys on {}:'.format(host.slug))
+                    filename = "/tmp/deployer-tempfile-{}-{}".format(time.time(), random.randint(0, 1000000000))
```
Comment (jonathanslenders) [inline deployer/shell.py:384]: Don't create temporary files. You don't need them.

File: deployer/shell.py [hunk 2]
```diff
@@ -357,6 +357,65 @@ class RunWithSudo(Run):
...
+                    try:
+                        # Check if authorizedkeysfile is defined in host
+                        if hasattr(host, 'authorizedkeysfile'):
+                            authfilename = host.expand_path(host.authorizedkeysfile)
+                        else:
+                            authfilename = host.expand_path("~/.ssh/authorized_keys")
+
+                        # Read file
+                        if host.exists(authfilename):
+                            host.get_file(authfilename, filename)
```
Comment (jonathanslenders) [inline deployer/shell.py:395]: You don't have to call `get_file`, you can read a file to memory:

```
with host.open(authfilename) as f:
    for line in f: ...
```

**Comment by jonathanslenders** [comment]
By the way, I'm already thinking a while about allowing to add `--`-extensions as a user of the library. We'll get there, but first I have to finish this library and use `pyline` for command line handling: https://github.com/jonathanslenders/cmdline/

**Comment by Duologic** [comment]
I've created a node in our deployer code. Thanks for the tips.

**Pull request closed without merging**